### PR TITLE
Issue8 pycsw distributor

### PR DESCRIPTION
**Pull request by mortenwh**
**Summary**: pycsw distributor with tests

**Related issue**: #8

**Suggested reviewer(s)**: @vkbo 

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

**Comment by johtoblan** [comment]
Python 3.8 ser ut til å gjøre A is B ekvivalent med A == B. Derfor feiler testen før 3.8

File: dmci/distributors/pycsw_dist.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    TOTAL_DELETED = 'total_deleted'
+    TOTAL_INSERTED = 'total_inserted'
+    TOTAL_UPDATED = 'total_updated'
+    STATUS = [TOTAL_DELETED, TOTAL_INSERTED, TOTAL_UPDATED]
+
+    def __init__(self, cmd, xml_file=None, metadata_id=None, **kwargs):
+
+        super().__init__(cmd, xml_file, metadata_id, **kwargs)
+
+    def run(self):
```
Comment (vkbo) [inline dmci/distributors/pycsw_dist.py:22]: docstring missing

File: dmci/distributors/pycsw_dist.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        ======
+        status : boolean
+            True if the transaction succeeded, otherwise False
+        """
+        if key not in self.STATUS:
+            logger.error("Input key must be '%s', '%s' or '%s'" % (self.TOTAL_INSERTED,
+                                                                   self.TOTAL_UPDATED,
+                                                                   self.TOTAL_DELETED))
+            return False
+        if not type(resp) is requests.models.Response:
```
Comment (vkbo) [inline dmci/distributors/pycsw_dist.py:132]: This is fine as long as you only accept exact type match. If the object can be inherited, you must instead use:
```python
if not isinstance(resp, requests.models.Response):
```

But regardless, the line should at least be rewritten to:
```python
if type(resp) is not requests.models.Response:
```
As @johtoblan  pointed out.

File: dmci/distributors/pycsw_dist.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            xml representation of the pycsw result
+
+        Return
+        ======
+        dict : status of insert, update and delete
+        """
+        if key not in self.STATUS:
+            logger.error("Input key must be '%s', '%s' or '%s'" % (self.TOTAL_INSERTED,
+                                                                   self.TOTAL_UPDATED,
+                                                                   self.TOTAL_DELETED))
```
Comment (vkbo) [inline dmci/distributors/pycsw_dist.py:158]: Just a suggestion for code format. This is much more readable:
```python
logger.error("Input key must be '%s', '%s' or '%s'" % (
    self.TOTAL_INSERTED, self.TOTAL_UPDATED, self.TOTAL_DELETED)
)
```

File: tests/conftest.py [hunk 4]
```diff
@@ -48,3 +50,137 @@ def refDir():
...
+    text = ('<?xml version="1.0" encoding="UTF-8" standalone="no"?>'
+            '<!-- pycsw 2.7.dev0 -->'
+            '<SomeNewTagName>'
+            '<TransactionSummary>'
+            '<totalInserted>0</totalInserted>'
+            '<totalUpdated>0</totalUpdated>'
+            '<totalDeleted>1</totalDeleted>'
+            '</TransactionSummary>'
+            '</SomeNewTagName>')
+    return [id, text]
```
Comment (vkbo) [inline tests/conftest.py:186]: I think it is better if these are defined in the pycsw test file, as they are specific to that module?

If so, the scope should be changed to "module"

**Review by vkbo — CHANGES_REQUESTED** [review]
I will push some changes for the superclass run() function in a different PR as I cannot run the tests here locally for some reason. Don't have the time to figure out why.

File: dmci/distributors/pycsw_dist.py (hunk 1 above)
Comment (mortenwh) [reply dmci/distributors/pycsw_dist.py:22]: solved in 3da4ca046dd032a97e2df21b42d8b6fe1e77cf5a

File: dmci/distributors/pycsw_dist.py (hunk 2 above)
Comment (mortenwh) [reply dmci/distributors/pycsw_dist.py]: Solved in 29ab426b602c87447e96f6ffa09585a049c715d5

File: dmci/distributors/pycsw_dist.py (hunk 3 above)
Comment (mortenwh) [reply dmci/distributors/pycsw_dist.py:166]: Solved in e0d9e2b758513483ef47ba55dc16eef4f4704ff9

File: tests/conftest.py (hunk 4 above)
Comment (mortenwh) [reply tests/conftest.py]: Done in 069863dcb3295e17eab877e60c9c4b34692ed20d

**Pull request merged**